Documentation/designs/status-subresource.md

```diff
@@ -0,0 +1,436 @@
+## Background
+
+The status subresource is a well-defined concept in Kubernetes:
+* [Kubernetes documentation](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#status-subresource) about custom resource definitions.
+* [Kubernetes API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties).
+* [kubebuilder](https://book-v1.book.kubebuilder.io/basics/status_subresource.html) documentation.
+* [OperatorSDK](https://sdk.operatorframework.io/docs/building-operators/golang/advanced-topics/#manage-cr-status-conditions) documentation.
+
+As much as possible, the proposal follows the recommendations established by the Kubernetes community.
+
+The feature is tracked in [#3335](https://github.com/prometheus-operator/prometheus-operator/issues/3335).
+
+## API
+
+### Prometheus
+
+The Prometheus CRD has a `Status` subresource that exposes the following fields:
+* `Paused`
+* `Replicas`
+* `UpdatedReplicas`
+* `AvailableReplicas`
+* `UnavailableReplicas`
+
+We propose to add new fields:
+* `Conditions` as recommended by the document describing the [Kubernetes API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties).
+* `ShardStatuses` which is a drilled-down version for each Prometheus shard.
+* [TBD] Number of resources (service monitors, pod monitors, probes and Prometheus rules) selected/rejected by the operator.
+
+```golang
+type PrometheusStatus struct {
+	// Represents whether any actions on the underlying managed objects are
+	// being performed. Only delete actions will be performed.
+	Paused bool `json:"paused"`
+	// Total number of non-terminated pods targeted by this Prometheus deployment
+	// (their labels match the selector).
+	Replicas int32 `json:"replicas"`
+	// Total number of non-terminated pods targeted by this Prometheus deployment
+	// that have the desired version spec.
+	UpdatedReplicas int32 `json:"updatedReplicas"`
+	// Total number of available pods (ready for at least minReadySeconds)
+	// targeted by this Prometheus deployment.
+	AvailableReplicas int32 `json:"availableReplicas"`
+	// Total number of unavailable pods targeted by this Prometheus deployment.
+	UnavailableReplicas int32 `json:"unavailableReplicas"`
+	// The current state of the Prometheus service.
+	// +patchMergeKey=type
+	// +patchMergeStrategy=merge
+	// +optional
+	Conditions []PrometheusStatusCondition `json:"conditions,omitempty"`
+	// The list has one entry per shard. Each entry provides a summary of the shard status.
+	// +patchMergeKey=shardID
+	// +patchMergeStrategy=merge
+	// +optional
+	ShardStatuses []ShardStatus `json:"shardStatuses,omitempty"`
+}
+
+// PrometheusStatusCondition represents the state of the resources associated with the Prometheus resource.
+// +k8s:deepcopy-gen=true
+type PrometheusStatusCondition struct {
```

Comment:
nit: afaik kubernetes use notation as follows: `<obj>` + `Condition`, so in this case, it would be `PrometheusCondition`.